models/record.go

```diff
@@ -328,6 +338,20 @@ func (r Records) GroupedByLabel() ([]string, map[string]Records) {
 	return order, groups
 }
 
+// GroupedByFQDN returns a map of keys to records, grouped by FQDN.
+func (r Records) GroupedByFQDN() ([]string, map[string]Records) {
```

Comment:
This will group records of different types together. I'm wondering what use case there is for this? I haven't needed to do that before, so wondering if this is too specific to gandi to belong in the models package? Usually want to group by the name/type tuple.

Reply:
I think that if one provider needs it, multiple will probably need it.  Maybe it should be in the diff package?